src/main/java/com/yelp/nrtsearch/server/luceneserver/doc/LoadedDocValues.java

```diff
@@ -494,4 +500,64 @@ public int size() {
       return values.size();
     }
   }
+
+  public static final class SingleVector extends LoadedDocValues<VectorType> {
+    private final BinaryDocValues docValues;
+    private VectorType value;
+
+    public SingleVector(BinaryDocValues docValues) {
+      this.docValues = docValues;
+    }
+
+    /**
+     * Set method to set the lucene level doc id to lookup value from index and initialize the
+     * loaded doc value index by loading vector data
+     */
+    public void setDocId(int docID) throws IOException {
+      if (docValues.advanceExact(docID)) {
+        value = decodeBytesRefToVectorType(docValues.binaryValue());
+      } else {
+        value = null;
+      }
+    }
+
+    /** Decodes binary doc value to float array and wraps it into a VectorType */
+    private static VectorType decodeBytesRefToVectorType(BytesRef bytesRef) {
+      byte[] byteArray = bytesRef.bytes;
+      float[] floats = new float[byteArray.length / Float.BYTES];
+      FloatBuffer fb = ByteBuffer.wrap(byteArray, 0, byteArray.length).asFloatBuffer();
```

Comment:
You will want to use the `BytesRef` properties here to ensure you only use valid data. i.e. `float[bytesRef.length / Float.BYTES]` and `ByteBuffer.wrap(bytesRef.bytes, bytesRef.offset, bytesRef.length)`